Nirs actionnetwork changes

NirTatcher opened the pull request:
Added some get functions for ActionNetwork module, I have more functions to add (create and update functions) after this batch is approved.
Please let me know if it looks ok :)

shaunagm commented:
Hi @NirTatcher sorry it's taking so long to get to this. I plan to review next week!

NirTatcher commented:
Hi @shaunagm, the PR didn't pass the linting tests, I fixed it (it should now pass the linting tests), updated the branch by the latest commit in main branch.
Thank you!

shaunagm commented on parsons/action_network/action_network.py:
```diff
@@ -51,7 +51,7 @@ def _get_entry_list(self, object_name, limit=None, per_page=25, filter=None):
             response = self._get_page(object_name, page, per_page, filter=filter)
             page = page + 1
-            response_list = response["_embedded"][f"osdi:{object_name}"]
+            response_list = response["_embedded"][list(response["_embedded"])[0]]
```
Can you explain what's going on here? I want to make sure this isn't a breaking change.

shaunagm replied:
Thank you for the explanation! Super helpful. So it sounds like this will not break any existing code, because only `object_name`s like "donations" was working anyway, and your approach still works for them. This just expands the valid object_names. And I didn't realize this before, but `object_name` really only seems to be used internally, and is not exposed to the end-user, so that makes worrying about breaking changes way less important anyway. :)

shaunagm commented on test/test_action_network/test_action_network.py:
```diff
@@ -348,6 +348,46 @@ def setUp(self, m):
             },
```
You've added a number of methods with no test coverage. I realize that having test coverage is of relatively marginal value given that we have to mock the endpoints, and I don't want to create a ton of busywork for you, but I would love it if we could check that the call is what we expect, given the parameters input.

NirTatcher replied:
Hi @shaunagm,
I reorganized the code and added tests for all functions added by me. 
Please let me know if it looks ok to you.

shaunagm replied:
These look great! Thank you for adding all these tests, Nir.

NirTatcher commented:
> Overall this looks great, had one quick question and a request about test coverage. Sorry for the delay in reviewing.

Hi @shaunagm,

I reorganized the code and added tests for every function. Everything should be working fine now please let me know how it looks like to you.

shaunagm reviewed (approved):
These look great. Thanks so much for adding all these methods, @NirTatcher!

shaunagm commented:
Merged! Thank you so much @NirTatcher
